util/package_watcher.go

```diff
@@ -0,0 +1,83 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package util
+
+import (
+	"log"
+	"os"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/radovskyb/watcher"
+)
+
+const watcherPollingPeriod = time.Second
+
+var (
+	w *watcher.Watcher
+
+	watchedPackages []Package
+)
+
+func MustUsePackageWatcher(packagePaths []string) {
+	log.Println("Use package watcher")
+
+	var err error
+	watchedPackages, err = getPackagesFromFilesystem(packagePaths)
+	if err != nil {
+		log.Println(errors.Wrap(err, "watcher error: reading packages failed"))
+	}
+
+	w = watcher.New()
+	w.SetMaxEvents(1)
+
+	for _, p := range packagePaths {
+		err = w.AddRecursive(p)
+		if err != nil && !os.IsNotExist(err) {
+			log.Fatal(errors.Wrapf(err, "watching directory failed (path: %s)", p))
+		}
+	}
+
+	go func() {
+		go w.Start(watcherPollingPeriod)
```

Comment:
Why is this started inside the go routine? Could this go on line 42?

Reply:
Actually, it blocks. I can move the whole line up: `go w.Start(...)`

Reply:
jep, that is what I mean. I just saw routine in routine which set off alarm bells :-)

Reply:
Fixed :)